src/trace.rs

```diff
@@ -0,0 +1,81 @@
+use bson::Bson;
+
+pub(crate) mod command;
+
+pub(crate) const COMMAND_TRACING_EVENT_TARGET: &str = "mongodb::command";
+
+trait TracingRepresentation {
+    fn tracing_representation(self) -> String;
+}
+
+impl TracingRepresentation for bson::oid::ObjectId {
+    fn tracing_representation(self) -> String {
+        self.to_hex()
+    }
+}
+
+impl TracingRepresentation for bson::Document {
+    fn tracing_representation(self) -> String {
+        Bson::Document(self).into_relaxed_extjson().to_string()
+    }
+}
+
+impl TracingRepresentation for crate::error::Error {
+    fn tracing_representation(self) -> String {
+        self.to_string()
+    }
+}
+
+/// We don't currently use all of these levels but they are included for completeness.
+#[allow(dead_code)]
+pub(crate) enum TracingOrLogLevel {
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+}
+
+impl TracingOrLogLevel {
+    pub(crate) const fn as_log_level(&self) -> log::Level {
+        match self {
+            TracingOrLogLevel::Error => log::Level::Error,
+            TracingOrLogLevel::Warn => log::Level::Warn,
+            TracingOrLogLevel::Info => log::Level::Info,
+            TracingOrLogLevel::Debug => log::Level::Debug,
+            TracingOrLogLevel::Trace => log::Level::Trace,
+        }
+    }
+
+    pub(crate) const fn as_tracing_level(&self) -> tracing::Level {
+        match self {
+            TracingOrLogLevel::Error => tracing::Level::ERROR,
+            TracingOrLogLevel::Warn => tracing::Level::WARN,
+            TracingOrLogLevel::Info => tracing::Level::INFO,
+            TracingOrLogLevel::Debug => tracing::Level::DEBUG,
+            TracingOrLogLevel::Trace => tracing::Level::TRACE,
+        }
+    }
+}
+
+/// Emits a tracing event at debug level. In tests, the provided client ID will be added as field to
+/// each event to enable distinguishing which client produced the event. Outside of tests, the ID
+/// will be ignored.
```

Comment:
I wonder if including client ID in non-tests scenarios would actually be useful too, since it would allow a user to distinguish between clients if they happen to use multiple in their applications. I feel like we may have discussed this before (and I apologize if that is the case!), but is there a reason why we wouldn't want this in the actual driver?

Reply:
1. I don't think its strictly necessary to include in the public API right now, though it could definitely be useful. I'd lean towards leaving it out for now in case something like this gets stabilized across drivers.
2. An ObjectID / UUID would also work fine and not require synchronization across client instances. As an aside, I wonder if we could use the Topology ID for this? We already publish it for TopologyOpening events, and AFAIK there is only one generated per client's lifetime.

Reply:
good idea re: the topology ID. it does appear it is generated just once when the `Topology` is initialized. 
I refactored to use that instead, and to add it to events unconditionally (i.e. not based on whether we are in tests). 

I named it `topology_id`, because I figured that's the name we'll go with when including it in SDAM tracing events, and this will make it more obvious that it's the same type of ID and can be used to correlate SDAM events with other events.

Reply:
Given that we're using topology ID now (which every driver should have), would that be something to consider upstreaming into the spec as a requirement?

Reply:
good question; I started a thread in the spec channel about this.